### DISCUSSION TITLE
Flex vs. Stack Documentation

### DISCUSSION THREAD

**Discussion by bhainesva** [chakra-ui/chakra-ui#8641] [Q&A] [answered]
### Description

Hello, I'm trying to understand the difference between the `HStack` and `Flex` components. 

The "[Notes on Stack vs Flex](https://v2.chakra-ui.com/docs/components/stack#notes-on-stack-vs-flex)" say that:
> The Stack component and the Flex component have their children spaced out evenly but the key difference is that the Stack won't span the entire width of the container whereas the Flex will. Another thing to note is that the items in both Stack and Flex are aligned in the center by default.

But when using the components, Flex doesn't seem to span the full width as described: https://codesandbox.io/p/sandbox/chakra-ui-flex-vs-stack-263q33?file=%2Fsrc%2Findex.tsx%3A20%2C1

The [Flex and Spacer vs Grid vs Stack](https://v2.chakra-ui.com/docs/components/flex#flex-and-spacer-vs-grid-vs-stack) similarly says:
> In HStack, the children will have equal spacing between them but they won't span the entire width of the container.

In the example given this is true, but the difference seems to be because of the use of `<Spacer />` in the Flex example. When passed the same children the two components seem to render the same as far as I can tell.  

From looking at the code I think the only difference might be that `Stack` aliases some of the flex properties (spacing -> gap, justify -> justifyContent, etc. ) and accepts a `divider` prop. But I wanted to check I wasn't missing anything and ask whether the documentation should be updated if they are actually the same. 

### Link to Reproduction

https://codesandbox.io/p/sandbox/chakra-ui-flex-vs-stack-263q33?file=%2Fsrc%2Findex.tsx%3A20%2C1

### Steps to reproduce

1. Go to https://codesandbox.io/p/sandbox/chakra-ui-flex-vs-stack-263q33?file=%2Fsrc%2Findex.tsx%3A20%2C1
2. See that HStack and Flex render mostly the same

### Chakra UI Version

2.8.2

### Browser

Firefox 127.0.1 (64-bit)

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by gcb** [comment]
flex does expand on 3.x

**Comment by segunadebayo** [comment, marked as answer]
v2 docs were misleading. Stack is Flex plus `gap`.
